tests/unit_tests/test_engine.py

```
@@ -367,3 +387,1790 @@ def test_add_health_check_not_callable(test_app: Engine):
+
+            # Next call should inspect again
+            test_app.notify("test2")
+            assert mock_sig.call_count == 2
+
+    def test_cache_stores_correct_values(self, test_app: Engine):
+        """Test that cache stores correct boolean values for different notifiers."""
+
+        def legacy_notifier(_message: str) -> None:
+            pass
+
+        def modern_notifier(_message: str, attachments: list[Attachment] | None = None) -> None:  # noqa: ARG001
```

Ruff (0.14.13)

Unused `noqa` directive (unused: `ARG001`)

Remove unused `noqa` directive

(RUF100)